lib/src/room.dart

```diff
@@ -425,36 +422,61 @@ class KurentoRoom extends ChangeNotifier implements Room {
     }
   }
 
+  Future<void> _getServiceRequestStatus() async {
+    Map<String, String?> response = await _client.getServiceRequestStatus(_serviceRequest.id);
+
+    await _updateServiceRequestStatus(response['status']!, response['agentFirstName']);
+  }
+
   Future<void> _handleServiceRequestPresenceMessage(String message) async {
     Map<String, dynamic> json = jsonDecode(message);
-    if (json['id'] != _serviceRequest.id) {
-      // Ignore status updates for other service requests (right now, Platform enforces that an Explorer can only be in
-      // one session, but that may change in the future).
-      return;
-    }
+    if (json['id'] != _serviceRequest.id) return;
 
-    _log.info('service request status changed id=${json['id']} status=${json['status']}');
+    await _updateServiceRequestStatus(json['status'], json['agentFirstName']);
+  }
+
+  Future<void> _updateServiceRequestStatus(String status, String? agentFirstName) async {
+    _log.info('updating service request status=$status');
+
+    switch (status) {
+      case 'REQUEST':
+        // Nothing to do.
+        break;
 
-    switch (json['status']) {
       case 'ASSIGNED':
-        _serviceRequestState = ServiceRequestState.assigned;
-        _agentName = json['agentFirstName'];
+        if (_serviceRequestState == ServiceRequestState.queued) {
+          _serviceRequestState = ServiceRequestState.assigned;
+          _agentName = agentFirstName;
+          notifyListeners();
+        }
         break;
 
       case 'STARTED':
-        _serviceRequestState = ServiceRequestState.started;
+        if (_serviceRequestState == ServiceRequestState.queued) {
+          // HACK: If we still think we're queued, we missed a message. Notify listeners that we're assigned, and let
+          // our periodic timer change this to started the next time it fires.
+          _log.warning('missed service request status message topic=$_serviceRequestPresenceTopic');
+          await _updateServiceRequestStatus('ASSIGNED', agentFirstName);
```

Comment:
This means we'll be a little delayed in getting to `started`, but we will have transitioned states correctly.

Reply:
This logic is complicated to understand. I'd need more time to figure out if there is no bug here, but you are usually pretty thorough, so instead of making the family wait me for dinner any longer, I thought I'd ask you the following questions:

I was wondering why, if we missed the 'ASSIGNED` state, don't we simply not call `_getServiceRequestStatus` here? We could wait for the result and then set the state to `STARTED`. This would be much simpler than relying on a timer.

My understanding on how the messages work is quite limited here. Waiting to call here might mean we won't be able to get the agent's name anymore (you say the name comes with the "ASSIGNE" message on a comment above). Not only calling every 3 seconds might miss that window as well, but that whole condition could be useless??? I feel a simple and efficient solution could ensure that the agent name is sent on 'ASSIGNED' and 'STARTED' event?

Reply:
@ipainchaud let's huddle in the morning, as I think it'll be easier to thoroughly explain how this works in person.

In short, if we're not receiving messages on the service request presence topic, we're not going to receive _any_ messages on that topic. In the early days of the Aira Explorer app, this is what would cause service requests to be stuck "Waiting for Agent" forever, even while the Agent was actually connected. We had "fixed" that with a hack that forced the state to `started` if we received the Agent's audio and were still `queued`, but that introduced another bug: because the Aira Explorer app now delays joining the room until the state is `assigned`, the Explorer never joins the room if messages are missed. 🤦🏾 Anyway, I'm sure you have some great ideas and I'll discuss them with you tomorrow.

Reply:
A few more notes to set up our discussion tomorrow:

* The service request presence topic is unreliable and we haven't yet been able to figure out why. One theory is that this is due to Platform publishing messages with [QoS](https://www.hivemq.com/blog/mqtt-essentials-part-6-mqtt-quality-of-service-levels/) 0 (i.e. at most once), so missed messages are expected. If so, this would affect all topics; we just notice it more for this one because it controls the very visible service request state transitions.
* The service request presence topic is only used for state transitions. We could eliminate it and use the periodic timer instead; however, I opted to keep the topic and use the periodic timer as a backup because the topic requires less resources and results in near realtime state transitions (when it works).
* I opted to cancel the periodic timer after the service request is `started`. This means that if we're not receiving messages from the service request presence topic and the Agent ends the call, the call won't end on the Explorer side until the Explorer ends the call. Since the Explorer won't get billed for those additional seconds, I thought this was acceptable (versus keeping the timer running for potentially multiple hours). 
* If we're not receiving messages from the topic, it will take at least 3 seconds (the timer duration) to transition from `assigned` to `started`. It typically takes Dash about 2-3 seconds to [start a service request](https://github.com/aira/dashboard/blob/925aef12e02cf06a3586bf988c0b3708c711e393/src/renderer/store/modules/dashboard.js#L199-L288), so this is not out of the ordinary.
* When we removed the 90-second limit for queued service requests, we got reports of Explorers waiting for 10's of minutes without their calls being answered. Most of those cases appeared to be due to missed messages, and this change should address those too.